File: test-fixtures/php-api/crypto.php
<?php

use phpseclib3\Crypt\AES;
use phpseclib3\Crypt\RSA;

// --- AES-256-GCM encryption (FIPS Approved) ---
function encrypt_aes_gcm(string $key, string $plaintext): array {
    $iv = openssl_random_pseudo_bytes(12);
    $tag = '';
    $ciphertext = openssl_encrypt($plaintext, 'aes-256-gcm', $key, OPENSSL_RAW_DATA, $iv, $tag);
    return ['iv' => $iv, 'data' => $ciphertext, 'tag' => $tag];
}

// --- AES-256-CBC encryption (FIPS Approved) ---
function encrypt_aes_cbc(string $key, string $plaintext): string {
    $iv = openssl_random_pseudo_bytes(16);
    return openssl_encrypt($plaintext, 'aes-256-cbc', $key, OPENSSL_RAW_DATA, $iv);
}

// --- 3DES (FIPS Deprecated) ---
function encrypt_3des(string $key, string $plaintext): string {
    $iv = openssl_random_pseudo_bytes(8);
    return openssl_encrypt($plaintext, 'des-ede3-cbc', $key, OPENSSL_RAW_DATA, $iv);
}

// --- SHA-256 hash (FIPS Approved) ---
function hash_sha256(string $data): string {
    return hash('sha256', $data);
}

// --- SHA-1 hash (FIPS Deprecated) ---
function hash_sha1(string $data): string {
    return hash('sha1', $data);
}

// --- MD5 hash (NOT FIPS Approved) ---
function hash_md5(string $data): string {
    return hash('md5', $data);
}

// --- HMAC-SHA256 (FIPS Approved) ---
function hmac_sha256(string $key, string $data): string {
    return hash_hmac('sha256', $data, $key);
}

// --- RSA key generation (FIPS Approved) ---
function generate_rsa_key(): OpenSSLAsymmetricKey {
    return openssl_pkey_new([
        'private_key_bits' => 2048,
        'private_key_type' => OPENSSL_KEYTYPE_RSA,
    ]);
}

// --- Sodium: Ed25519 signing (FIPS Approved) ---
function generate_signing_keypair(): string {
    return sodium_crypto_sign_keypair();
}

// --- Sodium: X25519 key exchange (NOT FIPS Approved) ---
function generate_box_keypair(): string {
    return sodium_crypto_box_keypair();
}

// --- Sodium: ChaCha20-Poly1305 (NOT FIPS Approved) ---
function encrypt_chacha(string $key, string $nonce, string $plaintext, string $ad): string {
    return sodium_crypto_aead_chacha20poly1305_encrypt($plaintext, $ad, $nonce, $key);
}

// --- Sodium: BLAKE2b (NOT FIPS Approved) ---
function generic_hash(string $data): string {
    return sodium_crypto_generichash($data);
}

// --- Sodium: Argon2 password hashing (NOT FIPS Approved) ---
function hash_password(string $password): string {
    return sodium_crypto_pwhash(
        32, $password, random_bytes(16),
        SODIUM_CRYPTO_PWHASH_OPSLIMIT_INTERACTIVE,
        SODIUM_CRYPTO_PWHASH_MEMLIMIT_INTERACTIVE
    );
}

// --- bcrypt password hashing (NOT FIPS Approved) ---
function hash_password_bcrypt(string $password): string {
    return password_hash($password, PASSWORD_BCRYPT);
}

File: test-fixtures/elixir-phoenix-app/crypto.ex
defmodule PhoenixApp.Crypto do
  @moduledoc """
  Cryptographic utilities for the Phoenix application.
  """

  # --- AES-256-GCM encryption (FIPS Approved) ---
  def encrypt_aes_gcm(key, plaintext) do
    iv = :crypto.strong_rand_bytes(12)
    {ciphertext, tag} = :crypto.crypto_one_time(:aes_256_gcm, key, iv, plaintext)
    {iv, ciphertext, tag}
  end

  # --- AES-128-GCM encryption (FIPS Approved) ---
  def encrypt_aes_128_gcm(key, plaintext) do
    iv = :crypto.strong_rand_bytes(12)
    {ciphertext, tag} = :crypto.crypto_one_time(:aes_128_gcm, key, iv, plaintext)
    {iv, ciphertext, tag}
  end

  # --- ChaCha20-Poly1305 (NOT FIPS Approved) ---
  def encrypt_chacha(key, plaintext) do
    iv = :crypto.strong_rand_bytes(12)
    :crypto.crypto_one_time(:chacha20_poly1305, key, iv, plaintext)
  end

  # --- 3DES (FIPS Deprecated) ---
  def encrypt_3des(key, plaintext) do
    iv = :crypto.strong_rand_bytes(8)
    :crypto.block_encrypt(:des_ede3_cbc, key, iv, plaintext)
  end

  # --- SHA-256 hash (FIPS Approved) ---
  def hash_sha256(data) do
    :crypto.hash(:sha256, data)
  end

  # --- SHA-512 hash (FIPS Approved) ---
  def hash_sha512(data) do
    :crypto.hash(:sha512, data)
  end

  # --- SHA-1 hash (FIPS Deprecated) ---
  def hash_sha1(data) do
    :crypto.hash(:sha, data)
  end

  # --- MD5 hash (NOT FIPS Approved) ---
  def hash_md5(data) do
    :crypto.hash(:md5, data)
  end

  # --- HMAC-SHA256 (FIPS Approved) ---
  def hmac_sha256(key, data) do
    :crypto.mac(:hmac, :sha256, key, data)
  end

  # --- HMAC-SHA512 (FIPS Approved) ---
  def hmac_sha512(key, data) do
    :crypto.mac(:hmac, :sha512, key, data)
  end

  # --- RSA key generation (FIPS Approved) ---
  def generate_rsa_key do
    :crypto.generate_key(:rsa, {2048, 65537})
  end

  # --- ECDH key generation (FIPS Approved) ---
  def generate_ecdh_key do
    :crypto.generate_key(:ecdh, :secp256r1)
  end

  # --- bcrypt password hashing (NOT FIPS Approved) ---
  def hash_password_bcrypt(password) do
    Bcrypt.hash_pwd_salt(password)
  end

  def verify_password_bcrypt(password, hash) do
    Bcrypt.verify_pass(password, hash)
  end

  # --- Argon2 password hashing (NOT FIPS Approved) ---
  def hash_password_argon2(password) do
    Argon2.hash_pwd_salt(password)
  end

  # --- PBKDF2 (FIPS Approved) ---
  def hash_password_pbkdf2(password) do
    Pbkdf2.hash_pwd_salt(password)
  end
end
